packages/fiori/src/Timeline.ts

```diff
@@ -111,6 +188,116 @@ class Timeline extends UI5Element {
 			: Timeline.i18nBundle.getText(TIMELINE_ARIA_LABEL);
 	}
 
+	get growsOnScroll(): boolean {
+		return this.growing === TimeLineGrowingMode.Scroll;
+	}
+
+	get timeLineEndDOM(): Element {
+		return this.shadowRoot!.querySelector(".ui5-time-line-end-marker")!;
+	}
+
+	get growingButtonIcon() {
+		return this.layout === TimelineLayout.Horizontal ? "process" : "drill-down";
+	}
+
+	get moreBtn(): HTMLElement | null {
+		const domRef = this.getDomRef();
+
+		if (this.growsWithButton && domRef) {
+			return domRef.querySelector<HTMLElement>(`#${this._id}-growingButton`);
+		}
+
+		return null;
+	}
+
+	get showBusyIndicatorOverlay() {
+		return !this.growsWithButton && this.loading;
+	}
+
+	get growsWithButton(): boolean {
+		return this.growing === TimeLineGrowingMode.Button;
+	}
+
+	onAfterRendering() {
+		if (this.growsOnScroll) {
+			this.observeTimeLineEnd();
+		}
+	}
+
+	onEnterDOM() {
+		this.growingIntersectionObserver = this.getIntersectionObserver();
+	}
+
+	onExitDOM() {
+		this.growingIntersectionObserver!.disconnect();
+		this.growingIntersectionObserver = null;
+		this.timeLineEndObserved = false;
+	}
+
+	observeTimeLineEnd() {
+		if (!this.timeLineEndObserved) {
+			this.getIntersectionObserver().observe(this.timeLineEndDOM);
+			this.timeLineEndObserved = true;
+		}
+	}
+
+	getIntersectionObserver(): IntersectionObserver {
+		if (!this.growingIntersectionObserver) {
+			this.growingIntersectionObserver = new IntersectionObserver(this.onInteresection.bind(this), {
+				root: document,
+				threshold: 1.0,
+			});
+		}
+
+		return this.growingIntersectionObserver;
+	}
+
+	onInteresection(entries: Array<IntersectionObserverEntry>) {
+		if (this.initialIntersection) {
+			this.initialIntersection = false;
+			return;
+		}
+
+		if (entries.some(entry => entry.isIntersecting)) {
+			debounce(this.loadMore.bind(this), GROWING_WITH_SCROLL_DEBOUNCE_RATE);
```

Comment:
We can use arrow function here

```ts
debounce(() => this.loadMore(), GROWING_WITH_SCROLL_DEBOUNCE_RATE);
```

Reply:
This suggestion is not relevant. The 'debounce' method expects two arguments: a function and a delay. Your suggestion to pass an arrow function does not meet the method's expectations.

Reply:
So, an arrow function is not a function ?